### ISSUE TITLE
When a React project changes the components of a router, it does not trigger the HMR and requires a manual refresh or restart

### ISSUE THREAD

**Issue by sdhushu** [farm-fe/farm#857]
### Steps to reproduce

1. pnpm create farm
2. choose react template
3. add react router
4. pnpm dev
5. replace the element of the route

### What is expected?

Triggers HMR page changes

### What is actually happening?

No HMR triggered, no change on page

### System Info

```Shell
System:
    OS: macOS 13.3
    CPU: (8) arm64 Apple M1
    Memory: 111.61 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.15.1 - /usr/local/bin/node
    Yarn: 1.22.18 - /usr/local/bin/yarn
    npm: 6.14.14 - /usr/local/bin/npm
    pnpm: 7.9.5 - /usr/local/bin/pnpm
  Browsers:
    Chrome: 120.0.6099.129
    Safari: 16.4
  npmPackages:
    @farmfe/core: ^0.14.18 => 0.14.18
```

**Comment by wre232114** [comment]
Related to #460

**Comment by wre232114** [comment]
fixed in `@farmfe/core@0.15.9`
